### PR TITLE
Add manifest file for concurrent segment search in 2.10.0

### PR DESCRIPTION
**Pull request by sohami**
### Description
Adding the manifest file for concurrent segment search with the respective settings needed to enable the feature

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by sohami** [comment]
@prudhvigodithi Can you please help review this ?

**Comment by prudhvigodithi** [comment]
Thanks @sohami, LGTM. Also can you update the cron entry to make sure these tests are called periodically, here is the sample PR https://github.com/opensearch-project/opensearch-build/pull/3696/files. Once added you can monitor the runs in jenkins https://build.ci.opensearch.org/blue/organizations/jenkins/integ-test/.
Adding @gaiksaya as she will be the release manager for the `2.10.0` release.

**Comment by sohami** [comment]
> Thanks @sohami, LGTM. Also can you update the cron entry to make sure these tests are called periodically, here is the sample PR https://github.com/opensearch-project/opensearch-build/pull/3696/files. Once added you can monitor the runs in jenkins https://build.ci.opensearch.org/blue/organizations/jenkins/integ-test/. Adding @gaiksaya as she will be the release manager for the `2.10.0` release.

@prudhvigodithi Have added the changes for jenkins file, but not sure if that is the right way to do it. Can you please look into it and confirm ?

File: jenkins/check-for-build.jenkinsfile [hunk 1]
```diff
@@ -17,6 +17,7 @@ pipeline {
             H 1 * * * %INPUT_MANIFEST=2.9.1/opensearch-2.9.1.yml;TARGET_JOB_NAME=distribution-build-opensearch;BUILD_PLATFORM=linux macos windows;BUILD_DISTRIBUTION=tar rpm deb zip
             H 1 * * * %INPUT_MANIFEST=2.10.0/opensearch-dashboards-2.10.0.yml;TARGET_JOB_NAME=distribution-build-opensearch-dashboards;BUILD_PLATFORM=linux windows;BUILD_DISTRIBUTION=tar rpm deb zip
             H 1 * * * %INPUT_MANIFEST=2.10.0/opensearch-2.10.0.yml;TARGET_JOB_NAME=distribution-build-opensearch;BUILD_PLATFORM=linux macos windows;BUILD_DISTRIBUTION=tar rpm deb zip
+            H 1 * * * %INPUT_MANIFEST=2.10.0/opensearch-2.10.0.yml;TEST_MANIFEST=2.10.0/opensearch-2.10.0-concurrent-search-test.yml;TARGET_JOB_NAME=distribution-build-opensearch-concurrent-search;BUILD_PLATFORM=linux macos windows;BUILD_DISTRIBUTION=tar rpm deb zip
             H 1 * * * %INPUT_MANIFEST=2.8.1/opensearch-dashboards-2.8.1.yml;TARGET_JOB_NAME=distribution-build-opensearch-dashboards;BUILD_PLATFORM=linux windows;BUILD_DISTRIBUTION=tar rpm deb zip
```
Comment (prudhvigodithi) [inline jenkins/check-for-build.jenkinsfile:21]: This should be as `TARGET_JOB_NAME=distribution-build-opensearch`, you can ignore the `BUILD_PLATFORM`.

Comment (sohami) [reply jenkins/check-for-build.jenkinsfile:21]: Thanks. updated

Comment (gaiksaya) [reply jenkins/check-for-build.jenkinsfile:21]: @prudhvigodithi @sohami  This should not be updated in check for build. Cronjob for integtest should be added directly in integ-test file here: https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/opensearch/integ-test.jenkinsfile

**Review by gaiksaya — CHANGES_REQUESTED** [review]

File: jenkins/check-for-build.jenkinsfile (hunk 1 above)
Comment (gaiksaya) [reply jenkins/check-for-build.jenkinsfile:21]: You can ignore adding any cronjob for now. I can take it up later. Please only proceed with adding the concerned manifest.
Thanks!

Comment (sohami) [reply jenkins/check-for-build.jenkinsfile]: removed the jenkins file change. @gaiksaya Can you help with adding the cron related change to run the test with custom manifest file. We want to see if there are any plugin test failures with concurrent search enabled

**Review by gaiksaya — APPROVED** [review]

File: jenkins/check-for-build.jenkinsfile (hunk 1 above)
Comment (gaiksaya) [reply jenkins/check-for-build.jenkinsfile]: Will do!

Comment (gaiksaya) [reply jenkins/check-for-build.jenkinsfile]: Created the cron here: https://github.com/opensearch-project/opensearch-build/pull/3857

**Pull request merged**